### PR TITLE
Add Course::Level model

### PR DESCRIPTION
**Pull request by kxmbrian**
This PR implements Course Levels as per https://github.com/Coursemology/coursemology2/wiki/EXP-Points-and-Levels.

### CODE REVIEW

File: db/migrate/20150206020132_create_course_levels.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class CreateCourseLevels < ActiveRecord::Migration
+  def change
+    create_table :course_levels do |t|
+      t.integer :course_id
+      t.integer :exp_threshold
```
Comment (allenwq) [inline db/migrate/20150206020132_create_course_levels.rb]: <code>null:false</code> ?

File: db/migrate/20150206020132_create_course_levels.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+
+      t.timestamps
```
Comment (allenwq) [inline db/migrate/20150206020132_create_course_levels.rb]: same here

File: app/models/course/level.rb [hunk 3]
```diff
@@ -0,0 +1,4 @@
+class Course::Level < ActiveRecord::Base
+  belongs_to :course, inverse_of: :levels
+  validates :exp_threshold, numericality: { greater_than: 0 }
```
Comment (allenwq) [inline app/models/course/level.rb]: Can exp_threshold equal to 0 ?

Comment (kxmbrian) [inline app/models/course/level.rb]: It has been disallowed. The reasons are written in the wiki.

**Comment by lowjoel** [comment]
@allenwq have a look.

File: db/migrate/20150206020132_create_course_levels.rb [hunk 4]
```diff
@@ -0,0 +1,10 @@
+class CreateCourseLevels < ActiveRecord::Migration
+  def change
+    create_table :course_levels do |t|
+      t.integer :course_id,                   null: false
+      t.integer :experience_points_threshold, null: false
+
+      t.timestamps
```
Comment (allenwq) [inline db/migrate/20150206020132_create_course_levels.rb]: <code>null: false </code> ?

**Comment by lowjoel** [comment]
Thanks @allenwq!

**Pull request merged**